### ISSUE TITLE
[Build] QNN EP on windows

### ISSUE THREAD

**Issue by mwangistan** [microsoft/onnxruntime#16595]
### Describe the issue

Can't build onnx using QNN EP on windows

### Target platform

Windows

### Build script

`build.bat --config RelWithDebInfo --use_qnn --qnn_home="qualcomm_ai_engine_direct.2.12.1.230626.Windows-AnyCPU.exe" --build_wheel --cmake_generator "Visual Studio 17 2022" --skip_submodule_sync --config Release --build_dir \build\Windows`

### Error / output

 'QnnInterface.h': No such file or directory

### Visual Studio Version

Visual studio 17 2022

**Comment by jywu-msft** [comment]
qnn_home should point to your QNN SDK install location (which should have include and target subdirectories)
the build error is complaining it can't find QNN header files.
can you confirm the contents of your QNN SDK directory?

**Comment by mwangistan** [comment]
Thanks for the response @jywu-msft . I was not using the correct path

I'm now running 
`build.bat --config RelWithDebInfo --use_qnn --qnn_home="C:\Qualcomm\AIStack\QNN\2.10.40.4"" --build_wheel --cmake_generator "Visual Studio 17 2022" --skip_submodule_sync --config Release --build_dir \build\Windows `
2023-07-12 13:02:02.5998673 [E:onnxruntime:, qnn_execution_provider.cc:287 

The build completes successfully but I notice some error related to QNN SetupBackend. Am I missing something else? 
`onnxruntime::QNNExecutionProvider::GetCapability] QNN SetupBackend failed qnn_backend_manager.cc:40 onnxruntime::qnn::QnnBackendManager::GetQnnInterfaceProviders Unable to load backend, error: filename is null or empty`

<img width="945" alt="Screenshot 2023-07-12 131359" src="https://github.com/microsoft/onnxruntime/assets/15102839/9c13b833-3f85-4549-a906-2ada34623fd7">

**Comment by jywu-msft** [comment]
re: #3, there's a bit more detail to add.

in addition to adding the lib location (I updated comment above to provide example of lib location),
you also need to make sure the skel file (e.g. libQnnHtpV68Skel.so) is in the PATH as well. one can copy that file to the QNN SDK lib directory (e.g. C:\Qualcomm\AIStack\QNN\2.12.1.230626\lib\aarch64-windows-msvc). 

I tested latest build from main and it seems to work okay.

```
import onnxruntime as ort
sess = ort.InferenceSession('mobilenet_quant.onnx', providers=['QNNExecutionProvider'], provider_options=[{'backend_path':'QnnHtp.dll'}])
```
